File: rb/features.rb
# Pagespeed SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module PagespeedFeatures
  def self.make_feature(name)
    case name
    when "base"
      PagespeedBaseFeature.new
    when "test"
      PagespeedTestFeature.new
    else
      PagespeedBaseFeature.new
    end
  end
end

File: py/pagespeed_sdk/features.py
# Pagespeed SDK feature factory

from pagespeed_sdk.feature.base_feature import PagespeedBaseFeature
from pagespeed_sdk.feature.test_feature import PagespeedTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PagespeedBaseFeature(),
        "test": lambda: PagespeedTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
